gliderpy/plotting.py

```diff
@@ -90,3 +90,42 @@ def plot_transect(
     ax.set_ylim(ax.get_ylim()[0], 0)
 
     return fig, ax
+
+
+@register_dataframe_method
+def plot_ctd(
+    df: pd.DataFrame,
+    var: str,
+    ax: plt.Axes = None,
+    color: str = None
+) -> tuple:
+    """Make a CTD profile plot of pressure vs property
+    depending on what variable was chosen.
+      
+    :param var: variable to plot against pressure
+    :param ax: existing axis to plot on (default: None)
+    :param color: color for the plot line (default: None)
+    :return: figure, axes
+    """
+    g = df.groupby(["longitude", "latitude"])
+    profile = g.get_group((list(g.groups)[0]))
```

Comment:
@FloraSauerbronn in these two lines line we are:

1. grouping the data in lon, lat poitns, so each group is a CTD profile;
2. getting the first group, aka, the first CTD profile.

A user may want to:
a. plot profile x by itself, not necessarily the first
b. plot all profiles for a quick inspection

For `a` we should introduce a new kw arg were the `[0]` above would  the profile number.
Option `b` is tricky b/c we can easily blow up the memory if the section is too big.

Can try to implement option `a`?

Reply:
Yes, I will try to come up with something today. I am still trying to understand why the pre-commit didn't work.

Reply:
You mean why pre-commit failed or why they did not work on your machine? Either way, don't worry about pre-commits until the last minute when the PR is ready to merge. We can use autofix to correct it then.

Reply:
Okay.

In the kwargs of the CTD plot, I would like the user to be able to choose which latitude and longitude they want to see a profile for. However, the problem is that they might not get the specific latitude or longitude they want. If the kwargs were just a simple index, I don't know how that would help because the user wouldn't know which latitude and longitude the profile corresponds to by just choosing an index in a dataframe. There needs to be a way for the user to see which latitudes and longitudes are available and then choose accordingly.

Reply:
Let's try something simpler first. A user already selected a bounding box and is aware of the positions where the profiles lies. So, accessing profiles based on integer indices is not a bad approach. You would need to make that `[0]` something like `[idx]` and add an `idx` argument with the accompanying documentation for it.

Reply:
Ok, feeling a bit weird about such a simple solution. I keep thinking about adding a title to the plot with the latitude and longitude, but the user can do this on their own.

Reply:
Doing the push again with changes..